src/rebar_prv_eunit.erl

```diff
@@ -42,13 +42,14 @@ do(State) ->
     ?DEBUG("Compiling EUnit instrumented modules in: ~p", [OutDir]),
     lists:foreach(fun(App) ->
                       AppDir = rebar_app_info:dir(App),
+                      AppOutDir = rebar_app_info:out_dir(App),
                       C = rebar_config:consult(AppDir),
                       S = rebar_state:new(State, C, AppDir),
                       %% combine `erl_first_files` and `eunit_first_files` and adjust
                       %% compile opts to include `eunit_compile_opts`, `{d, 'TEST'}`
                       %% and `{src_dirs, "test"}`
                       TestState = first_files(test_state(S, OutDir)),
-                      ok = rebar_erlc_compiler:compile(TestState, AppDir)
+                      ok = rebar_erlc_compiler:compile(TestState, AppDir, AppOutDir)
                   end, TestApps),
     ok = maybe_compile_extra_tests(TestApps, State, OutDir),
```

Comment:
this will currently generate an `ebin` dir in the root of the project (not in the root of it's `_build` location) but can easily be changed although i don't know where it would be best to put it right now

also filtering of checkapps needs to be changed to use `outdir` instead of `dir`